packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,207 @@ def apply_default_cascade(args):
+                argument, int
+            ):  # just a column name given as str or int
+                if not df_provided:
+                    raise ValueError(
+                        "String or int arguments are only possible when a"
+                        "DataFrame or an array is provided in the `data_frame`"
+                        "argument. No DataFrame was provided, but argument '%s'"
+                        "is of type str or int." % field
+                    )
+                # Check validity of column name
+                if argument not in args["data_frame"].columns:
+                    raise ValueError(
+                        "Value of '%s' is not the name of a column in 'data_frame'. "
+                        "Expected one of %s but received: %s"
+                        % (field, str(list(args["data_frame"].columns)), argument)
+                    )
+                if length and len(args["data_frame"][argument]) != length:
+                    raise ValueError(
+                        "All arguments should have the same length. "
+                        "The length of column argument `df[%s]` is %d, whereas the "
+                        "length of previous arguments %s is %d"
+                        % (
+                            field,
+                            len(args["data_frame"][argument]),
+                            str(list(df.columns)),
+                            length,
+                        )
+                    )
+                col_name = argument
+                if isinstance(argument, int):
+                    col_name = _name_heuristic(argument, field, reserved_names)
+                    if field_name not in array_attrables:
+                        args[field_name] = col_name
+                    else:
+                        args[field_name][i] = col_name
+                df[col_name] = args["data_frame"][argument]
+                continue
+            # Case of multiindex
+            elif isinstance(argument, pd.core.indexes.multi.MultiIndex):
+                raise TypeError(
+                    "Argument '%s' is a pandas MultiIndex."
+                    "pandas MultiIndex is not supported by plotly express" % field
+                )
+            # ----------------- argument is a column / array / list.... -------
+            else:
+                is_index = isinstance(argument, pd.core.indexes.range.RangeIndex)
+                # First pandas
+                # pandas series have a name but it's None
+                if (
+                    hasattr(argument, "name") and argument.name is not None
+                ) or is_index:
+                    col_name = argument.name  # pandas df
+                    if col_name is None and is_index:
+                        col_name = "index"
+                    if not df_provided:
+                        col_name = field
+                    else:
+                        keep_name = hasattr(
```

Comment:
instead of `hasattr`/`getattr` we could do `col_name in args["data_frame"].columns`/`args["data_frame"][col_name]` or something right?